### PR TITLE
Working version of gwpy-wav

### PR DESCRIPTION
**Pull request by andrew-lundgren**
A standalone command-line tool for grabbing some data from a channel and outputting a wave file. Can whiten or band-pass, and also change the output sample rate. Next I plan to implement linear pitch-shifting.

### CODE REVIEW

**Comment by duncanmmacleod** [comment]
@andrew-lundgren, thanks for this. Two questions

- why is the `TimeSeries.whiten` method not being used? If you have an alternate whitening method, we should be able to integrate it into `TimeSeries.whiten` as an option
- can the pitch shifting be made into a method of the `TimeSeries` object?

Also, please run `pep8 bin/gwpy-wav` against your code and patch to match the formatting guidelines.

**Comment by duncanmmacleod** [comment]
While I'm thinking about it, can the `wavefile.write` operation be integrated into `TimeSeries.write`?

**Comment by duncanmmacleod** [comment]
also, please run `pyflakes` to expose unused imports

**Comment by andrew-lundgren** [comment]
I want to apply band-passing and whitening at the same time, and then possibly pitch-shift in the frequency domain. It's easier to just write the operations out explicitly rather than use the gwpy whiten method.
We could probably make the wavfile.write part of TimeSeries. Should it include auto-scaling of the loudness?

File: bin/gwpy-wav [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+from numpy import roll
+from numpy.fft import rfft, irfft
+from scipy.signal import firwin, hann
+from scipy.io import wavfile
+
+__author__ = 'andrew lundgren'
+__email__ = 'andrew.lundgren@ligo.org'
+
+parser = argparse.ArgumentParser(description=__doc__, prog='gwpy-wav')
+parser.add_argument('-c', '--chan', type=str, required=True,
```
Comment (duncanmmacleod) [inline bin/gwpy-wav]: why are these options not arguments? i.e. use

```python
parser.add_argument('channel', help='Channel name')
```

Comment (andrew-lundgren) [inline bin/gwpy-wav]: Wouldn't that make them positional arguments? I've done my best to match the command-line options for gwpy-plot.

Comment (duncanmmacleod) [inline bin/gwpy-wav]: yes, positional arguments should be fine (we can make things work with LDVW with positional arguments)

**Comment by duncanmmacleod** [comment]
@andrew-lundgren, this PR has sat untouched for a while, so I'm going to close it. If you still think this tool would be a good addition to gwpy, please open a fresh PR.

**Comment by areeda** [comment]
@andrew-lundgren I'm working on a big re-factoring of LDVW that will require moving the Java code that creates .WAV files out of the tomcat servlets into a standalone program. I will look at making this part of LDVW.

**Pull request closed without merging**